wordpress-mobile/WordPress-Android
# My Site Dashboard: Tabs - Track tab tapped and tab shown events

Parent #16007 

This PR introduces the following track events:

- `my_site_tab_tapped` with property `tab=dashboard|site_menu`. Any time the user taps on a tab or swipes to another tab
- `my_site_site_menu_shown`. Any time the tab is shown - includes user gestures & other types of navigation to my site
- `my_site_dashboard_shown`. Any time the tab is shown - includes user gestures & other types of navigation to my site


**To test:**
- Launch the app
- If not enabled, please enable `MySiteDashboardTabsFeatureConfig`, then restart the app
- Navigate to My Site -> Site Menu tab
- Verify the logs contain:
 🔵 Tracked: my_site_tab_tapped, Properties: {"tab":"site_menu","default_tab_experiment":"dashboard|site_menu|nonexistent"}
 🔵 Tracked: my_site_site_menu_shown, Properties: {"default_tab_experiment":"dashboard|site_menu|nonexistent"}
- Tap the "Dashboard" tab
- Verify the logs contain:
🔵 Tracked: my_site_tab_tapped, Properties: {"tab":"dashboard","default_tab_experiment":"dashboard|site_menu|nonexistent"}
🔵 Tracked: my_site_dashboard_shown, Properties: {"default_tab_experiment":"dashboard|site_menu|nonexistent"}


## Regression Notes
1. Potential unintended areas of impact
Tabs are not tracked

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit testing

3. What automated tests I added (or what prevented me from doing so)
Add two tests to `MySiteViewModelTest`

PR submission checklist:
